src/dns/local.rs

```diff
@@ -0,0 +1,134 @@
+use std::collections::HashMap;
+use std::net::IpAddr;
+use wildmatch::WildMatch;
+use crate::dns::DNSEntry;
+
+/// Local override table that can be used instead of using /etc/hosts or similar 3rd party dns system.
+pub struct LocalOverrideTable {
+    /// Entries in the local override table. First iteration of the resolver: a simple list that
+    /// will be queried O(n). Later iterations will use a more efficient data structure like a tree
+    /// per domain part (e.g. com, org, net, etc.)
+    entries: Vec<DNSEntry>,
+    round_robin_ptrs: HashMap<String, usize>,
+}
+
+impl Default for LocalOverrideTable {
+    fn default() -> LocalOverrideTable {
+        LocalOverrideTable {
+            entries: Vec::new(),
+            round_robin_ptrs: HashMap::new(),
+        }
+    }
+}
+
+impl LocalOverrideTable {
+
+    pub fn resolve(&mut self, domain: &str) -> Option<IpAddr> {
+        let mut ips = Vec::new();
+        let mut wildcard_ips = Vec::new();
+
+        // For now it's ok to iterate the entries list, but later we should use a more efficient
+        // data structure.
+        for entry in &self.entries {
+            // We separate wildcard matches and specific matches.
+            if WildMatch::new(&entry.domain_name).matches(domain) {
+                wildcard_ips.push(entry.ip);
+            }
+            if entry.domain_name == domain {
+                ips.push(entry.ip);
+            }
```

Comment:
Here I would change it then (when we us a vec in DNSEntry)

```suggestion
            if entry.domain_name == domain {
                ips.append(entry.ips);
                break;
            } else if WildMatch::new(&entry.domain_name).matches(domain) { //when we have a more specific domain name don't use the WildMatch
                 wildcard_ips.append(entry.ips);
                 break;
            }

```